### DISCUSSION TITLE
Receiving participants from Pavlovia and sending them back to Prolific

### DISCUSSION THREAD

**Discussion by Adsecularis** [ctreffe/alfred#266] [Alfred Q&A] [answered]
Hello,

we wanna do a reaction time experiment with participants from Prolific. We plan to host the experiment itself on Pavlovia.
But we also have a questionnaire afterwards, which would be much easier to do over Alfred.

But the question is, how do we receive participants from Pavlovia in a way that automatically enters their Prolific ID?
Can Alfred read the query string of the URL the participants use to connect to the site?

And how do we automatically send them back to Prolific at the end?

And as a side question: Is Alfred yet able to handle accurate reaction time experiments? Because we would like to swtich to Alfred for that in the future.

**Comment by ctreffe** [comment, marked as answer]
Hey Jaron,

you can pass one or multiple parameter when starting an alfred3 experiment. Have a look at [urlargs](https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.experiment.ExperimentSession.urlargs.html#alfred3.experiment.ExperimentSession.urlargs) for  further information. It should be no problem to pass the Prolific ID from Pavlovia to alfred if Pavlovia can handle forwarding participants to an external url.

With alfred, you can easily receive participants from prolific and send them back with their ID and even status codes, if required. We conduct experiments using Prolific on a regular basis. As far as I know (I myself do not use Prolific), we use a simple [JavaScript element](https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.element.misc.JavaScript.html?highlight=javascript) to execute the forward to Prolific on the last experiment page.

Regarding reaction times, you might want to take a look at an early version of a [reaction time plugin](https://github.com/mate-code/alfred3-reaction-times). However, the plugin provides only basic functionality for displaying stimuli and recording reactions. We are currently not working on this topic, hence I am sorry to say that there is nothing else I can provide beyond the current version on GitHub. However, we are using the plugin regulary to display stimuli with accurate timing, and, from a technical standpoint, this works exceptionally well.
